functions/v2/firebase/Firebase.Tests/FirebaseFirestoreTest.cs

```diff
@@ -0,0 +1,131 @@
+using Xunit;
+
+namespace Firebase.Tests
+{
+    public class FirebaseFirestoreTest : FunctionTestBase<FirebaseFirestore.Function>
+    {
+        [Fact]
+        public async Task FullLogging()
+        {
+            string documentName = "projects/my-project/databases/(default)/documents/games/xyz";
+            var data = new DocumentEventData
+            {
+                OldValue = new Document
+                {
+                    Name = documentName,
+                    Fields =
+                    {
+                        { "highScore", new Value { IntegerValue = 10 } },
+                        { "name", new Value { StringValue = "Function Tester" } }
+                    }
+                },
+                Value = new Document
+                {
+                    Name = documentName,
+                    Fields =
+                    {
+                        { "highScore", new Value { IntegerValue = 20 } },
+                        { "name", new Value { StringValue = "Function Tester" } }
+                    }
+                }
+            };
+
+            ByteString eventDataByteString;
+            using (MemoryStream stream = new MemoryStream())
+            {
+                CodedOutputStream output = new CodedOutputStream(stream);
+                data.WriteTo(output);
+                output.Flush();
+                eventDataByteString = ByteString.CopyFrom(stream.ToArray());
+            }
+
+            var wrappedData = new Google.Protobuf.WellKnownTypes.Any()
+            {
+                TypeUrl = "types.googleapis.com/" + DocumentEventData.Descriptor.FullName,
+                Value = eventDataByteString
+            };
+
+            CloudEvent cloudEvent = new CloudEvent
+            {
+                Id = "0",
+                Subject = "documents/games/xyz",
+                Data = wrappedData.ToByteArray(),
+                Type = DocumentEventData.UpdatedCloudEventType,
+                Source = new Uri("https://example.com")
+            };
+
+            await ExecuteCloudEventRequestAsync(cloudEvent);
+
+            var logEntries = GetFunctionLogEntries();
+
+            var expectedMessages = new[]
+            {
+                "Function triggered by event on documents/games/xyz",
+                "Event type: google.cloud.firestore.document.v1.updated",
+                "Old value: highScore: 10, name: Function Tester",
+                "New value: highScore: 20, name: Function Tester"
+            };
+            var actualMessages = GetFunctionLogEntries().Select(entry => entry.Message).ToArray();
+
+            Assert.Equal(expectedMessages, actualMessages);
+        }
+
+        [Fact]
+        public async Task MissingFieldsIgnored()
+        {
+            // There really would normally be Value or OldValue, but it's simpler to test
+            // a completely empty payload.
+            var data = new DocumentEventData();
+
+            ByteString eventDataByteString;
```

Comment:
You should be able to use just `var wrappedData = Any.Pack(data);` - although it's not clear to me why there's an `Any` involved in the first place.